roles/k8s-gpu-plugin/tasks/main.yml

```diff
@@ -19,24 +20,37 @@
     name: openshift
     virtualenv: "{{ deepops_venv }}"
   when: ansible_distribution == 'Ubuntu'
+  environment: "{{proxy_env if proxy_env is defined else {}}}"
 
 - name: install openshift python client for k8s_raw module
   yum:
     name: python2-openshift
   when: ansible_os_family == 'RedHat'
+  environment: "{{proxy_env if proxy_env is defined else {}}}"
+
+- name: pull GPU device plugin file
+  get_url:
+    url: "{{ nvidia_k8s_device_plugin_def }}"
+    dest: "{{ nvidia_k8s_device_plugin_file }}"
+    mode: 0755
+  environment: "{{proxy_env if proxy_env is defined else {}}}"
 
 - name: create GPU device plugin
   k8s:
     state: present
-    definition: "{{ lookup('url', nvidia_k8s_device_plugin_def, split_lines=False) }}"
+    definition: "{{ lookup('file', nvidia_k8s_device_plugin_file, split_lines=False) }}"
```

Comment:
This task failed in my test because the `lookup` runs on the Ansible control machine, not on the remote machine it's configuring. If performing the `get_url` on the remote machine, you need to separately register the contents of the file you downloaded, rather than using a lookup.

I was able to run successfully in my test using the following change:

```
--- a/roles/k8s-gpu-plugin/tasks/main.yml
+++ b/roles/k8s-gpu-plugin/tasks/main.yml
@@ -35,10 +35,14 @@
     mode: 0755
   environment: "{{proxy_env if proxy_env is defined else {}}}"

+- name: get contents of device plugin definition
+  command: "cat {{ nvidia_k8s_device_plugin_file }}"
+  register: nvidia_k8s_device_plugin_contents
+
 - name: create GPU device plugin
   k8s:
     state: present
-    definition: "{{ lookup('file', nvidia_k8s_device_plugin_file, split_lines=False) }}"
+    definition: "{{ nvidia_k8s_device_plugin_contents.stdout }}"
     ## definition: "{{ lookup('url', nvidia_k8s_device_plugin_def, split_lines=False) }}"
   run_once: true
   environment:
```